runtime/src/main/java/io/quarkiverse/rabbitmqclient/RabbitMQClientProducer.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkiverse.rabbitmqclient;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import io.quarkus.arc.DefaultBean;
+import io.quarkus.runtime.TlsConfig;
+
+/**
+ * Producer class for creating in injecting a {@link RabbitMQClient} instance.
+ *
+ * @author b.passon
+ */
+@Singleton
+public class RabbitMQClientProducer {
+
+    private RabbitMQClientImpl rabbitMQClient;
+
+    /**
+     * Creates a singleton {@link RabbitMQClient}.
+     *
+     * @param config the {@link RabbitMQClientConfig} to use.
+     * @param tlsConfig the {@link TlsConfig} to use.
+     * @return a configured {@link RabbitMQClient}.
+     */
+    @DefaultBean
+    @Singleton
+    @Produces
+    public RabbitMQClient rabbitMQClient(RabbitMQClientConfig config, TlsConfig tlsConfig) {
+        if (rabbitMQClient == null) {
+            rabbitMQClient = new RabbitMQClientImpl(config, tlsConfig);
+        }
+        return rabbitMQClient;
+    }
+
+    /**
+     * Destroys the {@link RabbitMQClient} and closes all connections.
+     */
+    @PreDestroy
+    public void destroy() {
+        rabbitMQClient.disconnect();
+    }
```

Comment:
You should check for nullity here as the pre-destroy if for the producer but in case the RabittMQClient bean is removed from Arc (if the extension is here but the client not injected, it will not be produced) it will be null.

Reply:
Added null-check